pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -490,49 +482,73 @@ def phase_str(phi: float) -> str:
+                    sampled_seq.channel_samples[ch].phase[slot.ti : slot.tf]
+                    != 0
+                )
+                for slot in sampled_seq.channel_samples[ch].slots
             )
-            for pulse_num, seq_ in enumerate(seq._schedule[ch]):
-                # Select only `Pulse` objects
-                if isinstance(seq_.type, Pulse):
-                    if sampling_rate:
-                        area_val = (
-                            np.sum(yseff[0][seq_.ti : seq_.tf]) * 1e-3 / np.pi
+
+            for slot in sampled_seq.channel_samples[ch].slots:
+                if sampling_rate:
+                    area_val = (
+                        np.sum(yseff[0][slot.ti : slot.tf]) * 1e-3 / np.pi
+                    )
+                else:
+                    area_val = (
+                        np.sum(
+                            sampled_seq.channel_samples[ch].amp[
+                                slot.ti : slot.tf
+                            ]
                         )
-                    else:
-                        area_val = seq_.type.amplitude.integral / np.pi
-                    phase_val = seq_.type.phase
-                    x_plot = (seq_.ti + seq_.tf) / 2 / time_scale
-                    if (
-                        seq._schedule[ch][pulse_num - 1].type == "target"
-                        or not top
-                    ):
-                        y_plot = np.max(seq_.type.amplitude.samples) / 2
-                        top = True  # Next box at the top.
-                    elif top:
-                        y_plot = np.max(seq_.type.amplitude.samples)
-                        top = False  # Next box at the center.
-                    area_fmt = (
-                        r"A: $\pi$"
-                        if round(area_val, 2) == 1
-                        else rf"A: {area_val:.2g}$\pi$"
+                        * 1e-3
+                        / np.pi
                     )
-                    if not print_phase:
-                        txt = area_fmt
-                    else:
-                        phase_fmt = rf"$\phi$: {phase_str(phase_val)}"
-                        txt = "\n".join([phase_fmt, area_fmt])
-                    axes[0].text(
-                        x_plot,
-                        y_plot,
-                        txt,
-                        fontsize=10,
-                        ha="center",
-                        va="center",
-                        bbox=area_ph_box,
+                phase_val = sampled_seq.channel_samples[ch].phase[slot.tf - 1]
+                x_plot = (slot.ti + slot.tf) / 2 / time_scale
+                if (
+                    slot.ti
+                    in [
+                        target_slot.tf
+                        for target_slot in sampled_seq.channel_samples[
+                            ch
+                        ].target_time_slots
+                    ]
+                    or not top
+                ):
```

Comment:
Won't this miss target slots surrounded by delays?
It's true it's not very important, it's just to decide where to place the phase&area box. Perhaps if there is a delay, the box doesn't have to go down (the idea is just to avoid an overlap with the "target" box).

Still, this also seems very inefficient. Why not compute the list of `target_slot.tf`s once beforehand and use it here?

Reply:
> Won't this miss target slots surrounded by delays?
> It's true it's not very important, it's just to decide where to place the phase&area box. Perhaps if there is a delay, the box doesn't have to go down (the idea is just to avoid an overlap with the "target" box).

Yes, it may be drawn down if there is a delay; handling this case is not really straightforward and raises the doubt if the data we have in slots is enough

> Still, this also seems very inefficient. Why not compute the list of target_slot.tfs once beforehand and use it here?

You are right.

Reply:
Don't worry about eventual overlaps, I think it's not a big deal.